### PR TITLE
Product variant form script

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
Addressing issue #1228. Added some javascript to make the product variant form a bit more interactive and show the user what's happening when the "to order" checkbox is selected.

### CODE REVIEW

File: app/overrides/spree/admin/variants/_form/on_demand_script.html.haml.deface [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/ insert_bottom "[data-hook='admin_variant_form_fields']"
+:javascript
```
Comment (oeoeaio) [inline app/overrides/spree/admin/variants/_form/on_demand_script.html.haml.deface]: I'm not overly fussed about it, but seeing as we're doing everything else in coffeescript, might be worth doing this as :coffee instead?

File: app/overrides/spree/admin/variants/_form/on_demand_script.html.haml.deface [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+            } else {
+                if(on_hand.attr('data-stock') !== undefined) {
+                    on_hand.val(on_hand.attr('data-stock'));
+                } else {
+                    on_hand.val("0");
+                }
+            }
+
+        });
+
```
Comment (oeoeaio) [inline app/overrides/spree/admin/variants/_form/on_demand_script.html.haml.deface]: Just a few styling things. Could you remove the empty lines at the beginnings an ends of functions/if statements, and add a space between the `if`s and subsequent parentheses above. Sorry to be pedantic.

File: config/locales/en.yml [hunk 3]
```diff
@@ -140,6 +140,8 @@ en:
         inherits_properties?: Inherits Properties?
         available_on: Available On
         av_on: "Av. On"
+      variants:
+        to_order_tip: "Items made to order do not have a set stock level, such as loaves of bread made fresh to order."
```
Comment (oeoeaio) [inline config/locales/en.yml]: "On demand items do not have a set stock level...."

**Comment by Matt-Yorkley** [comment]
@oeoeaio All done. :)

**Comment by lin-d-hop** [comment]
@oeoeaio If you're now happy I'll pass to Oliver for testing.

**Comment by sstead** [comment]
Pushed.

**Comment by Matt-Yorkley** [comment]
@sstead Did this not get merged then?

**Comment by sstead** [comment]
Yes it got merged. Sorry am I not supposed to close it after it's been merged?

**Comment by Matt-Yorkley** [comment]
No that's fine, I was just confused.

**Pull request closed without merging**